gnocchi/carbonara.py

```diff
@@ -73,19 +76,23 @@ def __init__(self):
         super(InvalidData, self).__init__("Unable to unpack, invalid data")
 
 
+def datetime64_to_epoch(dt):
+    return (dt - UNIX_UNIVERSAL_START64) / ONE_SECOND
+
+
 def round_timestamp(ts, freq):
-    return pandas.Timestamp(
-        (pandas.Timestamp(ts).value // freq) * freq)
+    return UNIX_UNIVERSAL_START64 + numpy.floor(
```

Comment:
This is used for type conversion. The result on right side of `+` is a `numpy.timedelta64`, so in order to convert it, we need to add it to what is the 0 time indeed.